2009-06-09  Ulrich Drepper  <drepper@redhat.com>

	* libio/fileops.c (_IO_new_file_fopen): Don't prematurely stop
	parsing mode string.

--- libc/libio/fileops.c
+++ libc/libio/fileops.c
@@ -318,7 +318,7 @@ _IO_new_file_fopen (fp, filename, mode, 
 	  continue;
 	case 'c':
 	  fp->_flags2 |= _IO_FLAGS2_NOTCANCEL;
-	  break;
+	  continue;
 	default:
 	  /* Ignore.  */
 	  continue;
